### PR TITLE
Dev -> Master 2.4.0

### PR DESCRIPTION
**Pull request by sateeshperi**
### Pipeline Updates

- Updated template to nf-core/tools v2.8
- Add `--bamqc_regions_file` parameter for targeted methylation sequencing data #302
- ✨ Add NF-TEST pipeline end-to-end tests for existing CI tests #310 
- NF-TEST pipeline Assertions - ✨ Featuring NF-TEST Snapshots, `new File().exists()` and `!new File().exists()` checks
- 🛠️ `tests/pipeline/UTILS.groovy` for local groovy functions used in tests

### Bug fixes & refactoring

- 🛠️ update index file channels to explicit value channels #310
- 🐛 fix `params.test_data_base` in test and test_full configs #310
- 🤖 GitHub Actions CI - pull_request to `dev` tests with NXF_VER `latest-everything` #310
- 🤖 GitHub Actions CI - pull_request to `master` tests with NXF_VER `22.10.1` & `latest-everything` #310
- 🤖 GitHub Actions CI - `fail-fast` set to false #310
- 🐛 get to the bottom of index tests #278
- ✨ Support for Bismark methylation extraction `ignore` and `ignore_3prime` parameters when `ignore_r1` or `ignore_3prime_r1` are greater than 0. #322
- 🛠️ rename `ignore` -> `ignore_r1` and `ignore_3prime` -> `ignore_3prime_r1` params
- 🐛 fix `ignore_3prime_r2` param #299
- 🐛 removed unused directory #297

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -301,13 +301,27 @@
...
+                    "fa_icon": "far fa-eye-slash"
+                },
                 "ignore_r2": {
                     "type": "integer",
                     "default": 2,
                     "description": "Ignore methylation in first n bases of 5' end of R2",
                     "help_text": "Ignore the first <int> bp from the 5' end of Read 2 of paired-end sequencing results only. Since the first couple of bases in Read 2 of BS-Seq experiments show a severe bias towards non-methylation as a result of end-repairing sonicated fragments with unmethylated cytosines (see M-bias plot), it is recommended that the first couple of bp of Read 2 are removed before starting downstream analysis. Please see the section on M-bias plots in the Bismark User Guide for more details.",
                     "fa_icon": "far fa-eye-slash"
                 },
+                "ignore_3prime": {
```
Comment (SPPearce) [inline nextflow_schema.json:318]: And this be `ignore_3prime_r1`?

File: nextflow_schema.json [hunk 2]
```diff
@@ -301,13 +301,27 @@
...
+                    "type": "integer",
+                    "default": 0,
+                    "description": "Ignore methylation in last n bases of 3' end of R1",
+                    "help_text": "Ignore the first <int> bp from the 5' end of Read 2 of paired-end sequencing results only. Since the first couple of bases in Read 2 of BS-Seq experiments show a severe bias towards non-methylation as a result of end-repairing sonicated fragments with unmethylated cytosines (see M-bias plot), it is recommended that the first couple of bp of Read 2 are removed before starting downstream analysis. Please see the section on M-bias plots in the Bismark User Guide for more details.",
```
Comment (SPPearce) [inline nextflow_schema.json:322]: This text needs updating.

File: nextflow.config [hunk 3]
```diff
@@ -63,7 +65,9 @@ params {
     // Bismark default is 0.2 (L,0,-0.2), Bowtie2 default is 0.6 (L,0,-0.6)
     meth_cutoff                = null
     no_overlap                 = true
+    ignore                     = 0
```
Comment (SPPearce) [inline nextflow.config:68]: I don't like calling this parameter `ignore`, that seems far too generic a name.

**Review by SPPearce — CHANGES_REQUESTED** [review]
I don't think that `ignore` is a suitable name for a parameter, I think it should be `ignore_r1`. Similarly with `ignore_3prime` to make it match up.
Generally ok other than that, a few minor points. There are some #TODO strings, I don't know if that is because they are in the template or where they have come from.

File: nextflow_schema.json (hunk 1 above)
Comment (sateeshperi) [reply nextflow_schema.json:318]: good catch. thanks

File: README.md [hunk 4]
```diff
@@ -40,31 +40,45 @@ Choose between workflows by using `--aligner bismark` (default, uses bowtie2 for
...
+First, prepare a samplesheet with your input data that looks as follows:
 
-3. Download the pipeline and test it on a minimal dataset with a single command:
+`samplesheet.csv`:
 
-   ```bash
-   nextflow run nf-core/methylseq -profile test,YOURPROFILE --outdir <OUTDIR>
-   ```
+```csv
+sample,fastq_1,fastq_2,genome
```
Comment (SPPearce) [inline README.md:55]: Does the pipeline actually use the genome column currently? If not, remove from here.

**Review by SPPearce — APPROVED** [review]
Can you check the genome question, otherwise should be good to go.

File: nextflow_schema.json [hunk 5]
```diff
@@ -301,18 +301,32 @@
...
                     "fa_icon": "far fa-eye-slash"
                 },
                 "ignore_3prime_r2": {
                     "type": "integer",
                     "default": 2,
                     "description": "Ignore methylation in last n bases of 3' end of R2",
-                    "help_text": "Ignore the last <int> bp from the 3' end of Read 1 (or single-end alignment files) when processing the methylation call string. This can remove unwanted biases from the end of reads.",
+                    "help_text": "Ignore the last 2 bp from the 3' end of Read 1 (or single-end alignment files) when processing the methylation call string. This can remove unwanted biases from the end of reads.",
```
[the remark this answers is not in the document]
Comment (sateeshperi) [reply nextflow_schema.json:329]: ah you are right...fixing now

**Pull request merged**